File: user-service/src/test/java/io/ecommerce/userservice/config/docs/RestDocsConfiguration.java
package io.ecommerce.userservice.config.docs;

import org.apache.http.protocol.HTTP;
import org.springframework.restdocs.mockmvc.RestDocumentationResultHandler;
import org.springframework.restdocs.operation.preprocess.OperationRequestPreprocessor;
import org.springframework.restdocs.operation.preprocess.OperationResponsePreprocessor;
import org.springframework.restdocs.snippet.Snippet;

import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.document;
import static org.springframework.restdocs.operation.preprocess.Preprocessors.*;

/**
 * @author : choi-ys
 * @date : 2022/01/18 7:50 오후
 */
public interface RestDocsConfiguration {

    String HTTP = "http";
    String HTTPS = "https";
    String BASE_HOST = "ecommerce.io";
    String DEV_HOST = String.format("dev-user-service.%s", BASE_HOST);
    String STG_HOST = String.format("stg-user-service.%s", BASE_HOST);
    String PRD_HOST = String.format("prd-user-service.%s", BASE_HOST);

    static RestDocumentationResultHandler createDocument(Snippet... snippets) {
        return document("{class-name}/{method-name}",
                operationRequestPreprocessor(),
                operationResponsePreprocessor(),
                snippets
        );
    }

    private static OperationRequestPreprocessor operationRequestPreprocessor() {
        return preprocessRequest(
                modifyUris()
                        .scheme(HTTP)
                        .host(DEV_HOST)
                        .removePort(),
                prettyPrint());
    }

    private static OperationResponsePreprocessor operationResponsePreprocessor() {
        return preprocessResponse(
                modifyUris()
                        .scheme(HTTP)
                        .host(DEV_HOST)
                        .removePort(),
                prettyPrint());
    }
}

File: product-service/src/test/kotlin/io/ecommerce/productgservice/config/docs/RestDocsConfiguration.kt
package io.ecommerce.productgservice.config.docs

import org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.document
import org.springframework.restdocs.mockmvc.RestDocumentationResultHandler
import org.springframework.restdocs.operation.preprocess.OperationRequestPreprocessor
import org.springframework.restdocs.operation.preprocess.OperationResponsePreprocessor
import org.springframework.restdocs.operation.preprocess.Preprocessors
import org.springframework.restdocs.snippet.Attributes.Attribute
import org.springframework.restdocs.snippet.Snippet

/**
 * @author : choi-ys
 * @date : 2021-10-17 오전 4:18
 */
interface RestDocsConfiguration {

    companion object {
        const val HTTP: String = "http"
        const val HTTPS: String = "https"
        const val BASE_HOST: String = "whypie.me"
        const val DEV_HOST: String = "dev-project-api.$BASE_HOST"
        const val STG_HOST: String = "stg-project-api.$BASE_HOST"
        const val PRD_HOST: String = "prd-project-api.$BASE_HOST"

        fun format(value: String): Attribute = Attribute("format", value)

        fun createDocument(vararg snippets: Snippet): RestDocumentationResultHandler {
            return document("{class-name}/{method-name}",
                operationRequestPreprocessor(),
                operationResponsePreprocessor(),
                *snippets
            )
        }

        private fun operationRequestPreprocessor(): OperationRequestPreprocessor {
            return Preprocessors.preprocessRequest(
                Preprocessors.modifyUris()
                    .scheme(HTTP)
                    .host(DEV_HOST)
                    .removePort(),
                Preprocessors.prettyPrint()
            )
        }

        private fun operationResponsePreprocessor(): OperationResponsePreprocessor {
            return Preprocessors.preprocessResponse(
                Preprocessors.modifyUris()
                    .scheme(HTTP)
                    .host(DEV_HOST)
                    .removePort(),
                Preprocessors.prettyPrint()
            )
        }
    }
}